tests/nested/core20/boot-config-update/task.yaml

```diff
@@ -0,0 +1,57 @@
+summary: Check that a kernel refresh reseals
+
+systems: [ubuntu-20.04-64]
+
+prepare: |
+  # shellcheck source=tests/lib/nested.sh
```

Comment:
shouldn't this test have a preamble to prevent running a system where trust keys are not available?

i.e.

```bash
  if [ "$TRUST_TEST_KEYS" = "false" ]; then
      echo "This test needs test keys to be trusted"
      exit
  fi
```

Reply:
or maybe nested uc20 tests only run when trusting testkeys? I don't know for sure actually

Reply:
We do not need test keys, snapd we're refreshing from/to is unasserted.

Reply:
right but you can have a snapd that was built w/o testkeys when @sergiocazzolato does beta validation tests for example, he uses snapd from the beta channel which does not use the withtestkeys tag when it was built (for obvious reasons). But I'm not sure if we run the nested tests like this for the beta validation or if we just use the spread results from i.e. GCE where testkeys will have been included

Reply:
and iirc the bootassetstesting bit only works when snapd is built using the `withtestkeys` tag

Reply:
@sergiocazzolato do you run the nested tests as part of the validation?

Reply:
to be clear I think it's fine if we land this pr as-is, but we may need to revisit this when Sergio is back if nested tests are running with external backends as part of beta validation

Reply:
No worries, I'll prepare a followup with the test keys check and ping Sergio to take a look once he's back

Reply:
Opened #9904